msl_expressions/autogenerated/src/DomainBehaviour.cpp

```diff
@@ -1,125 +1,122 @@
+    }
+    this->passMsgPublisher = n.advertise<msl_helper_msgs::PassMsg>("WorldModel/PassMsg", 10);
+    this->watchBallMsgPublisher = n.advertise<msl_helper_msgs::WatchBallMsg>("/WorldModel/WatchBallMsg", 10);
+    this->debugMsgPublisher = n.advertise<msl_helper_msgs::DebugMsg>("/DebugMsg", 10);
 
-		__maxTranslation = (*sc)["Drive"]->get<double>("Drive", "MaxSpeed", NULL);
-	}
+    this->__maxTranslation = (*sc)["Drive"]->get<double>("Drive", "MaxSpeed", NULL);
+    this->minRotation = (*sc)["Actuation"]->get<double>("Dribble.MinRotation", NULL);
+    this->minRotationLeft = (*sc)["Actuation"]->get<double>("Dribble.MinRotationLeft", NULL);
+    this->minRotationRight = (*sc)["Actuation"]->get<double>("Dribble.MinRotationRight", NULL);
+    this->dribbleFactorRight = (*sc)["Actuation"]->get<double>("Dribble.DribbleFactorRight", NULL);
+    this->dribbleFactorLeft = (*sc)["Actuation"]->get<double>("Dribble.DribbleFactorLeft", NULL);
+}
 
-	DomainBehaviour::~DomainBehaviour()
-	{
-	}
+DomainBehaviour::~DomainBehaviour()
+{
+}
 
-	void alica::DomainBehaviour::send(msl_actuator_msgs::MotionControl& mc)
-	{
-//        this->wm->prediction.monitoring();
-		mc.senderID = ownID;
-		mc.timestamp = wm->getTime();
-		mc.motion.translation = min(__maxTranslation, mc.motion.translation);
-		motionControlPub.publish(mc);
-		wm->rawSensorData->processMotionControlMessage(mc);
-	}
+void alica::DomainBehaviour::send(msl_actuator_msgs::MotionControl &mc)
+{
+    //        this->wm->prediction.monitoring();
+    mc.senderID = ownID;
+    mc.timestamp = wm->getTime();
+    mc.motion.translation = min(this->__maxTranslation, mc.motion.translation);
+    this->motionControlPub.publish(mc);
+    this->wm->rawSensorData->processMotionControlMessage(mc);
+}
 
-	void alica::DomainBehaviour::send(msl_actuator_msgs::BallHandleCmd& bh)
-	{
-		supplementary::SystemConfig* sys = supplementary::SystemConfig::getInstance();
-		double minRotation = (*sys)["Actuation"]->get<double>("Dribble.MinRotation", NULL);
-		double minRotationLeft = (*sys)["Actuation"]->get<double>("Dribble.MinRotationLeft", NULL);
-		double minRotationRight = (*sys)["Actuation"]->get<double>("Dribble.MinRotationRight", NULL);
-		bh.enabled = true;
-		bh.senderID = ownID;
-		// this is only for nase and his new left motor
-		int sgnR = bh.rightMotor >= 0 ? 1 : -1;
-		int sgnL = bh.leftMotor >= 0 ? 1 : -1;
-		if (!(bh.rightMotor == 0) && !(bh.leftMotor == 0))
-		{
-			bh.rightMotor = (int)(max(minRotationRight, (abs(bh.rightMotor * 1.0)) / (*sys)["Actuation"]->get<double>("Dribble.DribbleFactorRight", NULL))) * sgnR;
-			bh.leftMotor = (int)(max(minRotationLeft, (abs(bh.leftMotor * 1.0)) / (*sys)["Actuation"]->get<double>("Dribble.DribbleFactorLeft", NULL))) * sgnL;
-		}
-		ballHandlePub.publish(bh);
-		
-	}
+void alica::DomainBehaviour::send(msl_actuator_msgs::BallHandleCmd &bh)
+{
+    bh.enabled = true;
+    bh.senderID = ownID;
+    // this is only for nase and his new left motor
```

Comment:
This comment still relevant?

Reply:
yes still relevant as long as nase has two different motors